metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetector.java

```diff
@@ -57,6 +57,15 @@
     metricQueryBuilder = new MetricQueryBuilder(asClient);
   }
 
+  MetricAnomalyDetector(Config appConfig, AttributeServiceClient asClient, QueryServiceClient qsc) {
+    queryServiceClient = qsc;
+    qsRequestTimeout =
```

Comment:
is `qsRequestTimeout` needed?

Reply:
Yes. over here
queryServiceClient.executeQuery()